core/src/main/java/org/apache/iceberg/deletes/EqualityDeleteWriter.java

```diff
@@ -56,10 +57,17 @@ public EqualityDeleteWriter(FileAppender<T> appender, FileFormat format, String
     this.equalityFieldIds = equalityFieldIds;
   }
 
+  @Override
+  public void write(T row) throws IOException {
+    appender.add(row);
+  }
+
+  @Deprecated
```

Comment:
I'll be up for that. I did not do this in this WIP PR to avoid touching more places. I think this is a low-level API which we can break.

Reply:
It's a low-level API, but it could be used by external projects like Hive since this is the easiest way to correctly write Iceberg files. I think we should deprecate them like Anton did here.